sympy/ntheory/factor_.py

```diff
@@ -2039,8 +2039,20 @@ class divisor_sigma(Function):
     def eval(cls, n, k=1):
         n = sympify(n)
         k = sympify(k)
+
         if n.is_prime:
             return 1 + n**k
+
+        if n.is_integer and not n.is_Integer:
+            p = []
+            for b, e in (_.as_base_exp() for _ in Mul.make_args(n)):
+                if b.is_prime and e.is_positive:
```

Comment:
If this fails then `None` will be returned? Is this expected? Any examples which can cause this to fail?

Reply:
No that same expression will be returned and this case is possible:

```
In[1]: k, m, n = symbols('k m n', integer=True)
In[2]: divisor_sigma(k*m)
Out[2]: divisor_sigma(k*m)


```